fem/bilininteg.cpp

```diff
@@ -901,6 +901,28 @@ void BoundaryMassIntegrator::AssembleFaceMatrix(
 }
 
 
+const IntegrationRule &ConvectionIntegrator::GetRule(const FiniteElement
+                                                     &trial_fe,
+                                                     const FiniteElement &test_fe,
+                                                     ElementTransformation &Trans)
+{
+   int order;
+   if (trial_fe.Space() == FunctionSpace::Pk)
+   {
+      order = trial_fe.GetOrder() + test_fe.GetOrder() - 2;
+   }
+   else
+   {
+      order = trial_fe.GetOrder() + test_fe.GetOrder() + trial_fe.GetDim() - 1;
+   }
+
+   if (trial_fe.Space() == FunctionSpace::rQk)
+   {
+      return RefinedIntRules.Get(trial_fe.GetGeomType(), order);
+   }
+   return IntRules.Get(trial_fe.GetGeomType(), order);
```

Comment:
The order/rule returned by this method is not consistent with the one used in `ConvectionIntegrator::AssembleElementMatrix` -- I think it is better to be consistent. The right formula should be (no explicit dependence on the `FunctionSpace`):
```c++
   order = Trans.OrderGrad(&trial_fe) + Trans.Order() + test_fe.GetOrder();
```

Reply:
Yes, that is much cleaner. Fixed.